python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,458 @@
+    with pytest.raises(TypeError, match=expected_expr):
+        pc.register_scalar_function(unary_scalar_function,
+                                    "output_function", doc, in_types,
+                                    None)
+
+    # validate input type
+    expected_expr = r'in_types must be a dictionary of InputType'
+    with pytest.raises(TypeError, match=expected_expr):
+        pc.register_scalar_function(unary_scalar_function,
+                                    "input_function", doc, None,
+                                    out_type)
+
+
+def test_varargs_function_validation():
+    def n_add(ctx, *values):
+        base_val = values[:2]
+        res = pc.call_function("add", base_val)
+        for other_val in values[2:]:
+            res = pc.call_function("add", [res, other_val])
+        return res
+
+    in_types = {"array1": pc.InputType.array(pa.int64()),
+                "array2": pc.InputType.array(pa.int64())
+                }
+    doc = {"summary": "n add function",
+           "description": "add N number of arrays"
+           }
+    pc.register_scalar_function(n_add, "n_add", doc,
+                                in_types, pa.int64())
+
+    func = pc.get_function("n_add")
+
+    assert func.name == "n_add"
+    error_msg = "VarArgs function 'n_add' needs at least 2 arguments"
+    with pytest.raises(pa.lib.ArrowInvalid, match=error_msg):
+        pc.call_function("n_add", [pa.array([1, 10]),
+                                   ])
+
+
+def test_function_doc_validation():
+
+    def unary_scalar_function(ctx, scalar):
+        return pc.call_function("add", [scalar, 1])
+
+    # validate arity
+    in_types = {"scalar": pc.InputType.scalar(pa.int64())}
+    out_type = pa.int64()
+
+    # doc with no summary
+    func_doc = {
+        "description": "desc"
+    }
+
+    expected_expr = "Function doc must contain a summary"
+
+    with pytest.raises(ValueError, match=expected_expr):
+        pc.register_scalar_function(unary_scalar_function, "no_summary",
+                                    func_doc, in_types,
+                                    out_type)
+
+    # doc with no decription
+    func_doc = {
+        "summary": "test summary"
+    }
+
+    expected_expr = "Function doc must contain a description"
+
+    with pytest.raises(ValueError, match=expected_expr):
+        pc.register_scalar_function(unary_scalar_function, "no_desc",
+                                    func_doc, in_types,
+                                    out_type)
+
+    # doc with empty dictionary
+    func_doc = {}
+    expected_expr = "Function doc must contain a summary"
+    with pytest.raises(ValueError, match=expected_expr):
+        pc.register_scalar_function(unary_scalar_function, "empty_dictionary",
+                                    func_doc, in_types,
+                                    out_type)
+
+
+def test_non_uniform_input_udfs():
+
+    def unary_scalar_function(ctx, scalar1, array1, scalar2):
```

Comment:
Does it actually need to use another function than the ones defined above?

Reply:
Yes you're correct we can update the assertion and use the ternary function.